pkg/networkutils/network.go

```diff
@@ -454,12 +460,37 @@ func setupENINetwork(eniIP string, eniMAC string, eniTable int, eniSubnetCIDR st
 			return errors.Wrap(err, "eni network setup: failed to clean up old routes")
 		}
 
-		if err := netLink.RouteAdd(&r); err != nil {
-			if !isRouteExistsError(err) {
-				return errors.Wrapf(err, "eni network setup: unable to add route %s/0 via %s table %d", r.Dst.IP.String(), gw.IP.String(), eniTable)
-			}
-			if err := netlink.RouteReplace(&r); err != nil {
-				return errors.Wrapf(err, "eni network setup: unable to replace route entry %s", r.Dst.IP.String())
+		// in case of route dependency, retry few times
+		retry := 0
+		for {
+
+			if err := netLink.RouteAdd(&r); err != nil {
+
+				if isNetworkUnreachable(err) {
+					retry++
+					if retry > maxRetryRouteAdd {
+						log.Errorf("Failed to add route %s/0 via %s table %d",
+							r.Dst.IP.String(), gw.IP.String(), eniTable)
+						return errors.Wrapf(err,
+							"eni network setup: failed unable to add route %s/0 via %s table %d",
+							r.Dst.IP.String(), gw.IP.String(), eniTable)
+					}
+					log.Debugf("Not able to add route route %s/0 via %s table %d (attempt %d/%d)",
+						r.Dst.IP.String(), gw.IP.String(), eniTable,
+						retry, maxRetryRouteAdd)
+					time.Sleep(retryRouteAddInterval)
+				} else if isRouteExistsError(err) {
+					if err := netlink.RouteReplace(&r); err != nil {
+						return errors.Wrapf(err, "eni network setup: unable to replace route entry %s", r.Dst.IP.String())
+					}
+					break
```

Comment:
Maybe a debug log as well? Something like
`log.Debugf("Successfully replaced route to be %s/0", r.Dst.IP.String())`